miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -215,23 +212,30 @@ public Sample get(Long sampleId) throws IOException {
     authorizationManager.throwIfNotReadable(sample);
     return sample;
   }
-  
+
   @Override
   public Long create(Sample sample) throws IOException {
     loadChildEntities(sample);
     authorizationManager.throwIfNotWritable(sample);
     setChangeDetails(sample, true);
-    if (sample.getSampleAdditionalInfo() != null) { 
-      if (sample.getSampleAdditionalInfo().getSampleClass() == null 
+    if (sample.getSampleAdditionalInfo() != null) {
+      if (sample.getSampleAdditionalInfo().getSampleClass() == null
           || sample.getSampleAdditionalInfo().getSampleClass().getSampleCategory() == null) {
         throw new IllegalArgumentException("Sample class or category missing");
       }
       if (!Identity.CATEGORY_NAME.equals(sample.getSampleAdditionalInfo().getSampleClass().getSampleCategory())) {
         sample.getSampleAdditionalInfo().setParent(findOrCreateParent(sample));
         validateHierarchy(sample);
+        if (sample.getSampleAdditionalInfo().getGroupId() == null) {
+          sample.getSampleAdditionalInfo().setGroupId(sample.getSampleAdditionalInfo().getParent().getSampleAdditionalInfo().getGroupId());
+        }
+        if (sample.getSampleAdditionalInfo().getGroupDescription() == null) {
+          sample.getSampleAdditionalInfo()
+              .setGroupDescription(sample.getSampleAdditionalInfo().getParent().getSampleAdditionalInfo().getGroupDescription());
+        }
```

Comment:
No. It's inherited once at the time of creation. If you change the parent the child won't see it. If you set it to null in the child, it will be null.